Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -1,147 +1,297 @@
+  simpa only [Cardinal.mk_uLift] using this
+
+/-- If `E` and `K` are isomorphic as `F`-algebras, then they have the same `finSepDegree`
+over `F`. -/
+theorem finSepDegree_eq_of_equiv (i : E ≃ₐ[F] K) :
+    finSepDegree F E = finSepDegree F K := by
+  simpa only [Cardinal.toNat_lift] using congr_arg Cardinal.toNat
+    (sepDegree_eq_of_equiv F E K i)
+
+@[simp]
+theorem sepDegree_self : sepDegree F F = 1 :=
+  le_antisymm (Cardinal.le_one_iff_subsingleton.2 AlgHom.subsingleton)
+    (Cardinal.one_le_iff_ne_zero.2 (instNeZeroSepDegree F F).out)
+
+@[simp]
+theorem finSepDegree_self : finSepDegree F F = 1 := by
+  simp only [finSepDegree, sepDegree_self, Cardinal.one_toNat]
+
+@[simp]
+theorem sepDegree_bot : sepDegree F (⊥ : IntermediateField F E) = 1 := by
+  have := sepDegree_eq_of_equiv _ _ _ (botEquiv F E)
+  rwa [sepDegree_self, Cardinal.lift_one, ← Cardinal.lift_one.{u, v}, Cardinal.lift_inj] at this
+
+@[simp]
+theorem finSepDegree_bot : finSepDegree F (⊥ : IntermediateField F E) = 1 := by
+  simp only [finSepDegree, sepDegree_bot, Cardinal.one_toNat]
+
+@[simp]
+theorem sepDegree_top : sepDegree F (⊤ : IntermediateField F E) = sepDegree F E := by
+  simpa only [Cardinal.lift_id] using sepDegree_eq_of_equiv F _ E topEquiv
+
+@[simp]
+theorem finSepDegree_top : finSepDegree F (⊤ : IntermediateField F E) = finSepDegree F E := by
+  simp only [finSepDegree, sepDegree_top]
+
+section
+
+variable [Algebra E K] [IsScalarTower F E K]
+
+theorem sepDegree_bot' : Cardinal.lift.{v} (sepDegree F (⊥ : IntermediateField E K)) =
+    Cardinal.lift.{w} (sepDegree F E) :=
+  sepDegree_eq_of_equiv _ _ _ ((botEquiv E K).restrictScalars F)
+
+@[simp]
+theorem sepDegree_bot'' (K : Type v) [Field K] [Algebra F K] [Algebra E K] [IsScalarTower F E K] :
+    sepDegree F (⊥ : IntermediateField E K) = sepDegree F E := by
+  simpa only [Cardinal.lift_id] using sepDegree_bot' F E K
+
+@[simp]
+theorem finSepDegree_bot' : finSepDegree F (⊥ : IntermediateField E K) = finSepDegree F E := by
+  simpa only [Cardinal.toNat_lift] using congr_arg Cardinal.toNat (sepDegree_bot' F E K)
+
+@[simp]
+theorem sepDegree_top' : sepDegree F (⊤ : IntermediateField E K) = sepDegree F K := by
+  simpa only [Cardinal.lift_id] using sepDegree_eq_of_equiv _ _ _
+    ((topEquiv (F := E) (E := K)).restrictScalars F)
+
+@[simp]
+theorem finSepDegree_top' : finSepDegree F (⊤ : IntermediateField E K) = finSepDegree F K := by
+  simp only [finSepDegree, sepDegree_top']
+
+end
+
+/-- A random bijection between `Emb F E` and `E →ₐ[F] K` if `E = F(S)` such that every element
+`s` of `S` is integral (= algebraic) over `F` and whose minimal polynomial splits in `K`.
+Combined with `instNonemptyEmb`, it can be viewed as a stronger version of
+`IntermediateField.nonempty_algHom_of_adjoin_splits`. -/
+def embEquivOfAdjoinSplits {S : Set E} (hS : adjoin F S = ⊤)
+    (hK : ∀ s ∈ S, IsIntegral F s ∧ Polynomial.Splits (algebraMap F K) (minpoly F s)) :
```

Comment:
apparently you already opened `Polynomial`; in any case you can at least use dot notation